skcosmo/selection/CUR.py

```diff
@@ -0,0 +1,218 @@
+from abc import abstractmethod
+import numpy as np
+import scipy
+from scipy.sparse.linalg import eigs as speig
+
+from ._base import _BaseSelection
+
+class _CUR(_BaseSelection):
+    """
+    Super-class defined for CUR selection methods
+
+    Parameters
+    ----------
+    mixing : float
+        mixing parameter, as described in PCovR as `alpha`
+        stored in `self.mixing`
+    iterative: boolean
+        whether to orthogonalize the matrices after each selection
+    k : int
+        number of eigenvectors to compute the importance score with
+    matrix : ndarray of shape (n x m)
+        Data to select from -
+        Feature selection will choose a subset of the `m` columns
+        Samples selection will choose a subset of the `n` rows
+        s   tored in `self.A`
+    precompute : int
+        Number of selections to precompute
+    progress_bar : bool, callable
+        Option to include tqdm progress bar or a callable progress bar
+        implemented in `self.progress(iterable)`
+    tolerance : float
+        Threshold below which values will be considered 0
+        stored in `self.tol`
+    Y (optional) : ndarray of shape (n x p)
+        Array to include in biased selection when mixing < 1
```

Comment:
This should not be part of the base class, as for mixin above, but defined in the PCovCUR derived class